components/autocomplete/useAutocomplete.js

```diff
@@ -0,0 +1,27 @@
+import { useState } from 'react';
+
+const useAutocomplete = (multiple, initialSelectedItems = []) => {
+    const [selectedItems, setSelectedItems] = useState(initialSelectedItems);
+    const [inputValue, setInputValue] = useState(
+        multiple || initialSelectedItems.length === 0 ? '' : initialSelectedItems[0].label
+    );
+
+    const submit = (item) => {
+        setInputValue(multiple ? '' : item.label);
+        setSelectedItems((selected) => (multiple ? [...selected, item] : [item]));
+    };
+
+    const remove = (i) => {
+        setSelectedItems((selected) => [...selected.slice(0, i), ...selected.slice(i + 1)]);
```

Comment:
use a filter to remove it, you won't need to create copies then spread the array :)